torch/_inductor/kernel/flex_attention.py

```diff
@@ -2677,7 +2678,28 @@ def flex_attention_backward(*args, **kwargs):
 
     # Create delta which will is needed for the bwd's kernel
     grad_lse_exp2 = lowerings[aten.mul](grad_logsumexp, 1 / math.log(2))
+
+    # Upcast mul to bf16 if operands are both fp8, since elementwise fp8 mul op is not supported
+    # due to hardware limitation.
+    fp8_dtypes = (
+        torch.float8_e4m3fn,
+        torch.float8_e5m2,
+    )
+    upcast_from_fp8 = False
+    orig_fp8_dtype = None
+    if out.get_dtype() in fp8_dtypes and grad_out.get_dtype() in fp8_dtypes:
+        orig_fp8_dtype = out.get_dtype()
+        assert out.get_dtype() == grad_out.get_dtype(), "FP8 types must match"
+        upcast_from_fp8 = True
+        out =  lowerings[prims.convert_element_type](out, torch.bfloat16)
+        grad_out =  lowerings[prims.convert_element_type](grad_out, torch.bfloat16)
+
     mul_delta = lowerings[aten.mul](out, grad_out)
+
+    # If mul was upcasted from fp8 to bf16, we need to downcast it back to fp8.
+    if upcast_from_fp8:
+        mul_delta = lowerings[prims.convert_element_type](mul_delta, orig_fp8_dtype)
+
     delta = lowerings[aten.sum](mul_delta, axis=-1)
     delta = lowerings[aten.sub](delta, grad_lse_exp2)
```

Comment:
hmm separate from the error you're seeing, are you going to run into similar issues trying to run `aten.sum/sub` in triton with fp8 inputs? You may need to delay downcasting back to fp8 til after these ops.